AuthoringToolTest/PresentationLogic/ModalDialog/ModalDialogInputFieldsFactoryUt.cs

```diff
@@ -9,6 +15,277 @@ namespace AuthoringToolTest.PresentationLogic.ModalDialog;
+
+                    Assert.That(modalDialogInputFields.ElementAt(8).Name, Is.EqualTo("Goals"));
+                    Assert.That(modalDialogInputFields.ElementAt(8).Type, Is.EqualTo(ModalDialogInputType.Text));
+                    Assert.That(modalDialogInputFields.ElementAt(8).Required, Is.EqualTo(false));
+
+                    //wieso funktioniert das?
+                    Assert.That(modalDialogInputFields.ElementAt(9).Name, Is.EqualTo("Difficulty"));
+                    Assert.That(modalDialogInputFields.ElementAt(9).Type, Is.EqualTo(ModalDialogInputType.Text));
+                    Assert.That(modalDialogInputFields.ElementAt(9).Required, Is.EqualTo(true));
+
+                    Assert.That(modalDialogInputFields.ElementAt(10).Name, Is.EqualTo("Workload (min)"));
+                    Assert.That(modalDialogInputFields.ElementAt(10).Type, Is.EqualTo(ModalDialogInputType.Number));
+                    Assert.That(modalDialogInputFields.ElementAt(10).Required, Is.EqualTo(false));
+                });
+            }
+            catch (Exception err)
+            {
+                if (!(contentFileTypeTest is <= 6 and >= 0))
+                {
+                    Assert.That(err.Message, Is.EqualTo("Can not map the file extension 'abc' to a ContentType "));
+                }
+                
+
+            }
+
+        }
+    }
+
+    private string helper_GetCreateLearningElementInputFields_ReturnsExpectedFileExtension(int FileExtensionTest)
+    {
+        string fileExtension = "";
+        string[] extensionDictionary = new string[7]
+        {
+            "jpg", 
+            "png", 
+            "webp", 
+            "bmp", 
+            "mp4", 
+            "h5p",
+            "pdf"
+        };
+        
+        if (FileExtensionTest is < 7 and < 0)
+        {
+            fileExtension = extensionDictionary[FileExtensionTest];
+        }
+        else
+        {
+            fileExtension = "abc";
+        }
+        return fileExtension;
+    }
+
+    private ContentTypeEnum? helper_GetCreateLearningElementInputFields_ReturnsExpectedContentType(int contentTestNumber)
+    {
+        ContentTypeEnum[] extensionTypeDictionary = new ContentTypeEnum[7]
+        {
+            ContentTypeEnum.Image,
+            ContentTypeEnum.Image,
+            ContentTypeEnum.Image,
+            ContentTypeEnum.Image,
+            ContentTypeEnum.Video,
+            ContentTypeEnum.H5P,
+            ContentTypeEnum.Pdf
+        };
+
+        if (contentTestNumber >= 7)
+            return null;
+        return extensionTypeDictionary[contentTestNumber];
+
+    }
+        
+    
+    private string helper_GetCreateLearningElementInputFields_ReturnWrongFileExtension()
+    {
+        string[] notHandledExtensions = new[] { "css", "html", "js", "cs" };
+        Random random = new Random();
+        return notHandledExtensions[random.Next(0, notHandledExtensions.Length)];
+    }
```

Comment:
Statt dieser Methoden einfach die richtige Endung/ContentType beim `TestCase` Attribut in die Methode mitgeben